### DISCUSSION TITLE
Diagonal artifact

### DISCUSSION THREAD

**Discussion by RobbeJ-2260915** [Auburn/FastNoiseLite#135] [Q&A]
Hello everyone, I was trying to use Fast noise light to generate a height map for a world but I keep geeting a diagonal artifact thorugh my result. Does anybody know what causes this? The white square is another object I'm rendering for testing.
![image](https://github.com/Auburn/FastNoiseLite/assets/113914974/aab0f17f-b9f8-4e29-8fe7-5c50112ba145)
```
static std::vector<float> GenerateNoiseMap(unsigned int width, unsigned int height, float scale) {
	  std::vector<float> noiseMap(width * height);
  
	  if (scale <= 0) scale = 0.0001f;
  
	  FastNoiseLite noise;
	  noise.SetNoiseType(FastNoiseLite::NoiseType_Perlin);
	  noise.SetFrequency(0.01f);
	  
	  int index = 0;
	  for (int y = 0; y < height; y++) {
		  for (int x = 0; x < width; x++) {
			  float value = noise.GetNoise((float)(x / scale), (float)(y / scale));
			  value = (value + 1.0f) * 0.5f;
			  noiseMap[index++] = value;
		  }
	  }
	  return noiseMap;
 }
```

**Comment by chadefranklin** [comment]
What was the issue?

**Reply by RobbeJ-2260915** [reply]
There was a problem somewhat later the created an offset of 1 when going down a layer and that created the diagonal line
